### DISCUSSION TITLE
Save Code and Autorun at next startup

### DISCUSSION THREAD

**Discussion by mperino** [BIPES/BIPES#235] [General]
I'm used to saving code to the device as Main.py and having it run at the next startup.  
It's probably pretty obvious and I'm just missing it, but after running the block program how do you save it to the board so it will run disconnected?
It seems that we are using "Paste" mode, and it seems like it's posssible to save cut and paste code as a file:
https://forum.micropython.org/viewtopic.php?t=4592

**Comment by mperino** [comment]
I figured it out...
Under Files
Blocks to Code
Select the Program
Rename it as main.py in the top banner.

Not super intuitive and would be an awesome enhancement to have a button next to the XML save that is "Save to Board as main.py"

**Comment by jetannenbaum** [comment]
I made a modification to the file naming code. Look under BIPES in the left navigation and use the Project Info block.   Set the description to main and the file name will be main.py.  Otherwise it will default to code.bipes.py (or whatever in in your description.bipes.py).  You will have to flush the cache to get the latest code.  Hope that makes it a little easier

Jim
